frame/support/src/traits.rs

```diff
@@ -1438,6 +1441,9 @@ pub trait OnInitialize<BlockNumber> {
 	/// The block is being initialized. Implement to have something happen.
 	///
 	/// Return the non-negotiable weight consumed in the block.
+	/// NOTE: This function is called BEFORE ANY extrinsic in a block is applied,
+	/// including inherent extrinsics. Hence for instance `timestamp`
```

Comment:
We should not mention timestamp here. Timestamp is not a required pallet of a substrate based runtime

Reply:
I've rephrased the comment to make it conditional. I'd rather have a concrete example here and I suppose that `timestamp` is rather a common component in the runtime and highly likely to be a common mistake as well.

Fine with removing it as well if you don't like the new phrasing either.